pkg/services/ngalert/state/manager.go

```diff
@@ -99,16 +105,22 @@ func (st *Manager) Warm(ctx context.Context) {
 		for _, entry := range cmd.Result {
 			ruleForEntry, ok := ruleByUID[entry.RuleUID]
 			if !ok {
-				st.log.Error("rule not found for instance, ignoring", "rule", entry.RuleUID)
+				// TODO Should we delete the orphaned state from the db?
```

Comment:
I say "yes" - if there is any change history we're interested in it is captured in other places like the annotations table. This is just dead weight. Will we ever clean it up if we don't do it here?

Of course, don't solve it here.